game-logic.js

```diff
@@ -0,0 +1,367 @@
+// https://github.com/rocketacademy/basics-beat-that
+
+// 1. There are TWO players and players take turns.
+// 2. Each player roll TWO dice
+
+const PLAYER_ONE = 0;
+const PLAYER_TWO = 1;
+const LAST_PLAYER = PLAYER_TWO;
+const PLAYERS = 2;
+var CURRENT_PLAYER = 0;
+
+const PLAYER_WIN_COUNT = [0, 0];
+/**
+ * playerRolls
+ * An example outcome when both players throw dice
+ * playerRolls =  [
+ *                   [[1, 3]],
+ *                  [[6, 6]]
+ *                ];
+ */
+var playerRolls = [[], []];
+
+/**
+ * playersRollsOrderings
+ * An example outcome when both players throw dice
+ * playersRollsOrderings =   [
+ *                            [[1, 3],[3, 1]],
+ *                            [[6, 6]]
+ *                          ];
+ */
+var playersRollsOrderings = [[], []];
+
+/**
+ * playerOrderingChoice
+ * A example outcome when both players had chosen their ordering
+ * playerOrderingChoice =   [
+ *                            [[3, 1]],
+ *                            [[6, 6]]
+ *                          ];
+ */
+var playerOrderingChoice = [[], []];
+
+const ACTION_ROLL = 0;
+const ACTION_ORDER = 1;
+const ACTION_GAME_SET = 2;
+
+var CURRENT_ACTION = ACTION_ROLL;
+
+var PREV_ROUND_RESULT_DESCRIPTION = "";
```

Comment:
nice and clear global variables with good commenting for the more complicated ones (your three 2d arrays)

Reply:
i generally only use SCREAM_CASE for constants and not for global variables that still might change, but i know this is a discussion you had with kenrick before and there is no clear standard for this, so its really up to personal preference

Reply:
i may have confused globals with constants  :sweat_smile: